File: JH/DAY5/11722.cpp
#include <bits/stdc++.h>

using namespace std;

int n;
long long arr[1001];
long long dp[1001];

int main(){

    cin >> n;

    for (int i = 0; i <n; ++i) {
        cin >> arr[i];
        dp[i] = 1;
    }

    long long ans = 1;

    for (int i = 0; i <n; ++i) {
        for (int j = 0; j <i ; ++j) {
            if(arr[i]<arr[j]){
                dp[i]= max(dp[i],dp[j]+1);
            }
            if(ans<dp[i]){
                ans = dp[i];
            }
        }
    }
    cout << ans <<"\n";
}

File: AJM/DAY6/11722.py
#가장 긴 감소하는 부분 수열

n = int(input())
arr = list(map(int,input().split()))

dp = [1]*1000

for i in range(1,n):
    for j in range(i):
        if arr[i] < arr[j]:
            dp[i] = max(dp[i],dp[j]+1)
    
print(max(dp))
